src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IIsPatternExpression.cs

```diff
@@ -0,0 +1,480 @@
+                //         if (/*<bind>*/x is /*</bind>*/const int y) Console.WriteLine(y);        
+                Diagnostic(ErrorCode.ERR_InvalidExprTerm, "const").WithArguments("const").WithLocation(8, 39),
+                // CS1026: ) expected
+                //         if (/*<bind>*/x is /*</bind>*/const int y) Console.WriteLine(y);        
+                Diagnostic(ErrorCode.ERR_CloseParenExpected, "const").WithLocation(8, 39),
+                // CS1023: Embedded statement cannot be a declaration or labeled statement
+                //         if (/*<bind>*/x is /*</bind>*/const int y) Console.WriteLine(y);        
+                Diagnostic(ErrorCode.ERR_BadEmbeddedStmt, "const int y").WithLocation(8, 39),
+                // CS0145: A const field requires a value to be provided
+                //         if (/*<bind>*/x is /*</bind>*/const int y) Console.WriteLine(y);        
+                Diagnostic(ErrorCode.ERR_ConstValueRequired, "y").WithLocation(8, 49),
+                // CS1002: ; expected
+                //         if (/*<bind>*/x is /*</bind>*/const int y) Console.WriteLine(y);        
+                Diagnostic(ErrorCode.ERR_SemicolonExpected, ")").WithLocation(8, 50),
+                // CS1513: } expected
+                //         if (/*<bind>*/x is /*</bind>*/const int y) Console.WriteLine(y);        
+                Diagnostic(ErrorCode.ERR_RbraceExpected, ")").WithLocation(8, 50),
+                // CS0103: The name 'y' does not exist in the current context
+                //         if (/*<bind>*/x is /*</bind>*/const int y) Console.WriteLine(y);        
+                Diagnostic(ErrorCode.ERR_NameNotInContext, "y").WithArguments("y").WithLocation(8, 70),
+                // CS0168: The variable 'y' is declared but never used
+                //         if (/*<bind>*/x is /*</bind>*/const int y) Console.WriteLine(y);        
+                Diagnostic(ErrorCode.WRN_UnreferencedVar, "y").WithArguments("y").WithLocation(8, 49)
+            };
+
+            VerifyOperationTreeAndDiagnosticsForTest<IsPatternExpressionSyntax>(source, expectedOperationTree, expectedDiagnostics);
+        }
+
+        [Fact, WorkItem(19927, "https://github.com/dotnet/roslyn/issues/19927")]
+        public void TestIsPatternExpression_InvalidInDefaultParameterInitializer()
+        {
+            string source = @"
+using System;
+class X
+{
+    void M(string x = /*<bind>*/string.Empty is string y/*</bind>*/)
+    {    
+    }
+}
+";
+            string expectedOperationTree = @"
+IIsPatternExpression (OperationKind.IsPatternExpression, Type: System.Boolean) (Syntax: 'string.Empty is string y')
+  Expression: IFieldReferenceExpression: System.String System.String.Empty (Static) (OperationKind.FieldReferenceExpression, Type: System.String) (Syntax: 'string.Empty')
+  Pattern: IDeclarationPattern (Declared Symbol: System.String y) (OperationKind.DeclarationPattern) (Syntax: 'string y')
```

Comment:
Work tracked with #20313

---
In reply to: [122784648](https://github.com/dotnet/roslyn/pull/20276#discussion_r122784648) [](ancestors = 122784648)